manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -798,11 +798,40 @@ func (r *FybrikApplicationReconciler) GetAllModules() (map[string]*fappv1.Fybrik
 		return moduleMap, err
 	}
 	for ind := range moduleList.Items {
+		if !validateModule(&moduleList.Items[ind]) {
+			continue
+		}
 		moduleMap[moduleList.Items[ind].Name] = &moduleList.Items[ind]
 	}
 	return moduleMap, nil
 }
 
+// validate that the module can be selected:
+// 1. check the module validation status
+// 2. special processing for modules that support any connections from a specific catalog provider -
+// set the protocol value to an empty one (empty values stand for "*" and are not checked)
+func validateModule(module *fappv1.FybrikModule) bool {
+	if len(module.Status.Conditions) > 0 &&
+		module.Status.Conditions[ModuleValidationConditionIndex].Status == v1.ConditionFalse {
+		return false
+	}
+	for capabilityInd := range module.Spec.Capabilities {
```

Comment:
actually, these 2 loops are not validation, but Capabilities settings, pls consider to put them into a separate method.

Reply:
Done.